src/OrchardCore.Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -119,5 +124,29 @@ private string GetPattern(AutoroutePart part)
 
             return pattern;
         }
+
+        private string GenerateUniquePath(string path, AutoroutePart context)
+        {
+            var similarPaths = _session.QueryIndexAsync<AutoroutePartIndex>(o => o.ContentItemId != context.ContentItem.ContentItemId && o.Path.StartsWith(path)).List().GetAwaiter().GetResult();
```

Comment:
Why `StartsWith`?

What I would do if there is a matching path:
- Try to extract a version by seeking the last `-` and for a `int.TryParse` on the value after that
- If succeeds start with this number, otherwise start at `1`
- Do `while(true)` by checking `path-version` and increment `version` until there is no match.